src/utils/Cypher.ts

```diff
@@ -1,44 +1,38 @@
+import { CryptoSetting, SettingDB } from '../types/Setting';
+
 import BackendError from '../exception/BackendError';
-import Configuration from './Configuration';
 import Constants from './Constants';
-import CryptoConfiguration from '../types/configuration/CryptoConfiguration';
+import { LockEntity } from '../types/Locking';
+import LockingManager from '../locking/LockingManager';
+import SettingStorage from '../storage/mongodb/SettingStorage';
+import Utils from './Utils';
 import _ from 'lodash';
 import crypto from 'crypto';
 
 const IV_LENGTH = 16;
 const MODULE_NAME = 'Cypher';
 
 export default class Cypher {
-  private static configuration: CryptoConfiguration;
-
-  public static getConfiguration(): CryptoConfiguration {
-    if (!this.configuration) {
-      this.configuration = Configuration.getCryptoConfig();
-      if (!this.configuration) {
-        throw new BackendError({
-          source: Constants.CENTRAL_SERVER,
-          module: MODULE_NAME,
-          method: 'getConfiguration',
-          message: 'Crypto configuration is missing'
-        });
-      }
-    }
-    return this.configuration;
-  }
 
-  public static encrypt(data: string): string {
+  public static async encrypt(data: string, tenantID: string, former = false): Promise<string> {
```

Comment:
I do not really like the signature change here. A better signature would have been to just pass the key to use as an argument.

Reply:
Yes, I agree. But in a previous review you suggested putting the tenantID in the encrypt/decrypt and making a getCryptoKey private method that we only call in Cypher.  But even if I pass the cryptoKey (which contains both key, keyPropeties, formerKey, formerKeyProperties), I still have the same problem. Is it better to create a new type which would contain only key and keyProperties and pass that to the encrypt/decrypt?

Reply:
Fair enough. Then just rename the boolean argument to something more meaningful like `useFormerKey`.

Reply:
I renamed in PR 03. Crypto